### ISSUE TITLE
Validate by Mapper Type DAG Improvements

### ISSUE THREAD

**Issue by amywieliczka** [ucldc/rikolti#568]
- ✨ **DAG Feature** ✨ Add an offset parameter to the validate by mapper type dag.
- ✨ **Airflow Feature** ✨ Move solr and couch environment variables into airflow variables so Adrian, Christine, and Gabriela can easily and self-service switch between running validation jobs against solr/couch-stg and solr/couch-prd (how does this impact local development?)
- ✨ **DAG Feature** ✨ Add a copy s3 to google drive task to move validation reports to google drive
- 🐛 **Validation Feature** 🐛 Update rikolti.validator.validate so it reports out any specific records that are in Solr but missing from Rikolti
-  ✨ **Validation Feature** ✨ Validation add record-level introspection reports for rikolti records vs. solr records that include:  
    - for each key in the intersection of rikolti_record.keys() and solr_record.keys(), if the key is not already in the set of validatable fields, check that rikolti_record[key] == solr_record[key] and if not, add a line item to the validation report reporting on the field name and field value in both rikolti and solr. 
    - for each key in the disjunctive union of rikolti_record.keys() and solr_record.keys(), if the key is not already in the set of validatable fields, add a line item to the validation report including the field name, the field value for the data source that includes it, and a description stating the data source that is missing the field. 
- 🐛 **Validation Bug** 🐛 Fix rikolti.validator.validate so it adds records missing from solr to the validation file instead of the job's log output - maybe this is not a bug. 
- ⚙️ **Refactor Chore** ⚙️ Add dag run IDs or dates to folders within the rikolti_data folder, ex: rikolti_data/3433-2023_10_18/vernacular_metadata-3433-2023_10_18, hook this in to _data_src, _data_dest options - maybe create a RikoltiStorage class that handles s3 vs. file storage generically that can be used throughout Rikolti - effectively wrapping boto s3 client. once a RikoltiStorage class is in place, we can define in one place how to assign dates (date + time for everything? just date? date + count - ie: 2023_10_18_run_1, 2023_10_18_run_2) and how to reference dates (always use most recent unless otherwise specified - what kind of hook to build in order to specify?)
    - This would enable us to maintain multiple different Rikolti outputs rather than replacing everything in the various s3 folders
    - This would enable us to run the pipeline starting with a specified data source, so we wouldn't have to re-run the fetching part again. 
    - This would give better data tracing between metadata stored in s3 and dag runs.

**Comment by amywieliczka** [comment]
I'm keeping this all hanging out in the backlog in one big issue with a low priority, and as I pick things off of it, I'll make individual issues for that thing.
